src/main/java/com/khutircraftubackend/userprofile/UserProfileService.java

```diff
@@ -0,0 +1,72 @@
+package com.khutircraftubackend.userprofile;
+
+import com.khutircraftubackend.auth.AuthenticationService;
+import com.khutircraftubackend.auth.UserEntity;
+import com.khutircraftubackend.auth.UserRepository;
+import com.khutircraftubackend.auth.UserService;
+import com.khutircraftubackend.mail.EmailSender;
+import com.khutircraftubackend.userprofile.request.ChangeOfEmailRequest;
+import com.khutircraftubackend.userprofile.request.ConfirmEmailRequest;
+import com.khutircraftubackend.userprofile.response.ChangeOfEmailResponse;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+import java.security.Principal;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.Map;
+
+@RequiredArgsConstructor
+@Setter
+@Getter
+public class UserProfileService {
+
+    private final UserService userService;
+    private final UserRepository userRepository;
+    private final AuthenticationService authenticationService;
+    private final EmailSender emailSender;
+    private Map<String, EmailTokenInfo> confirmEmail;
```

Comment:
`confirmEmail` better suitable for a method name.
because of:
- first word is a verb (what to do ? to confirm)
- second word - email, which is noun.
- altogether they make an "avtion", what to do ? - to confirm email.

better name for this map would be:
- emailConfirmationOptions
- emailConfirmationProperties
- emailConfirmationMap